File: src/ios/SmartIdPlugin.swift
import Foundation
import SmartId

/// Please read the Capacitor iOS Plugin Development Guide
/// here: https://capacitorjs.com/docs/plugins/ios
@objc(SmartIdPlugin) class SmartIdPlugin: CDVPlugin {

  private let implementation = SmartIdPlugin()

  @objc(startSmartId:)
  func startSmartId(command: CDVInvokedUrlCommand) {
    SID.startLocation()

    let pluginResult = CDVPluginResult(status: CDVCommandStatus_OK, messageAs: "The plugin succeeded")
    self.commandDelegate!.send(pluginResult, callbackId: command.callbackId)
  }

   @objc(getRawData:)
   func getRawData(command: CDVInvokedUrlCommand) {
     let arguments = command.arguments
     var backendDomain: String = ""

     if arguments.count > 0 {
        backendDomain = String(describing: command.arguments[0])
     }

     SID.shared
     .getRawData(applicationBackendDomain: backendDomain)
     .onSuccess(success: {time, response in
        do {
            let jsonObject: [String: Any] = [
                "data": response
            ]

            let jsonData = try JSONSerialization.data(withJSONObject: jsonObject, options: [])
            let jsonString = String(data: jsonData, encoding: .utf8)
            let result = CDVPluginResult(status: CDVCommandStatus_OK, messageAs: jsonString)
            self.commandDelegate!.send(result, callbackId: command.callbackId)
        } catch {
            let errorResult = CDVPluginResult(status: CDVCommandStatus_ERROR, messageAs: "Error al crear el objeto JSON")
            self.commandDelegate!.send(errorResult, callbackId: command.callbackId)
        }
     })
     .onFailure(failure: {time, message, errorCode in
        let errorResult = CDVPluginResult(status: CDVCommandStatus_ERROR, messageAs: message)
        self.commandDelegate!.send(errorResult, callbackId: command.callbackId)
     })
     .start()
   }

}

File: src/android/SmartIdPlugin.java
package com.develsystems.smartidplugin;

import android.content.Context;

import org.apache.cordova.CordovaPlugin;
import org.apache.cordova.CallbackContext;
import com.develsystems.smartid.SmartId;
import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;
import org.json.*;

/**
 * This class echoes a string called from JavaScript.
 */
public class SmartIdPlugin extends CordovaPlugin {

    @Override
    public boolean execute(String action, JSONArray args, CallbackContext callbackContext) throws JSONException {
        if (action.equals("getRawData")) {
            String backendDomain = "";

            if (args.length() > 0) {
                backendDomain = args.getString(0);
            }

            this.getRawData(backendDomain, callbackContext);
        }
        if (action.equals("startSmartId")) {
            this.startSmartId(callbackContext);
            return true;
        }

        return false;
    }

    private void startSmartId(CallbackContext callbackContext) {
        System.out.println("startSmartId method successful.");
    }

    private void getRawData(String backendDomain, CallbackContext callbackContext) {
        final Context context = this.cordova.getActivity().getApplicationContext();

        SmartId
                .getInstance()
                .GetRawData(context, backendDomain)
                .onSuccess((time, response) -> {
                    final JSONObject json = new new JSONObject();
                    json.put("data", response);

                    callbackContext.success(json);
                })
                .onFailure((time, message, errorCode) -> {
                    callbackContext.error(message + " - " + message);
                })
                .start();
    }

}
